tower-buffer/src/worker.rs

```diff
@@ -157,30 +165,43 @@ where
 
         loop {
             match try_ready!(self.poll_next_msg()) {
-                Some(msg) => {
+                Some((msg, first)) => {
+                    let _guard = msg.span.enter();
                     if let Some(ref failed) = self.failed {
+                        tracing::trace!("notifying about worker failure");
                         let _ = msg.tx.send(Err(failed.clone()));
                         continue;
                     }
 
                     // Wait for the service to be ready
+                    tracing::trace!(
+                        resumed = !first,
+                        message = "worker received request; waiting for service readiness"
+                    );
                     match self.service.poll_ready() {
                         Ok(Async::Ready(())) => {
+                            tracing::debug!("processing request");
```

Comment:
Nit/TIOLI: I might consider changing this to something like:
```suggestion
                            tracing::debug!(service.ready = true, message = "processing request");
```
That way, if we add a similar `service.ready` boolean field to the event that we record when the service isn't ready, a hypothetical subscriber could track the state of the service.